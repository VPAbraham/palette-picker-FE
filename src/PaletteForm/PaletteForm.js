import React, { Component } from 'react';
import './PaletteForm.scss';

class PaletteForm extends Component {
  constructor() {
    super();
    this.state = {
      paletteName: '',
      projectName: '',
      error: ''
    }
  }

  submitForm = (e) => {
    e.preventDefault();
    if (!this.state.paletteName && !this.state.projectName) {
      this.setState({error: "Please fill out all inputs to log in."})
    } else {
      // send up to app state
    }
  }

  handleChange = (e) => {
    e.preventDefault();
    this.setState({[e.target.name]: e.target.value})
  }

  handleClick = (e) => {
    e.preventDefault();
    console.log('hello')
  }

  render() {
    return(
      <form className="palette-form">
        <h2>PALETTE NAME</h2>
        <input
          className="palette-name-input"
          type="text"
          placeholder="Enter Palette Name"
          name="paletteName"
          value={this.state.paletteName}
          onChange={this.handleChange}
        />
        <button className="save-palette" onClick={(e) => this.submitForm(e)}> SAVE </button>
        <h2>SELECT PROJECT</h2>
        <div className="dropdown">
          <button className="drop-menu">Select Project</button>
          <div className="dropdown-content">
            <a onClick={this.handleClick}>Link 1</a>
          </div>
        </div>
      </form>
    )
  }
}

export default PaletteForm;
