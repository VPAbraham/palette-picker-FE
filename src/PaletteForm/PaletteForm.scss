@import url('https://fonts.googleapis.com/css?family=Montserrat&display=swap');

.palette-form {
  display: flex;
  flex-direction: column;
  margin-left: 3em;
  width: 50%;
}

h2 {
  color: rgb(189, 189, 189);
  display: flex;
  font-family: 'Montserrat', sans-serif;
  font-size: 1em;
  letter-spacing: 15px;
  line-height: 55px;
  text-align: left;
}

input {
  padding: 1em;
}

button {
  margin: 2em 0 2em 0;
  padding: 1em;
  width: 25%
}
