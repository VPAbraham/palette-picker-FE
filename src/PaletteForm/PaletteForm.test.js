import React from 'react';
import { PaletteForm } from './PaletteForm';
import { shallow } from 'enzyme';

describe('ProjectForm', () => {
  let wrapper;
  beforeEach(() => {
    const projects = [{ id: 20, name: 'Fall Colors' }, { id: 21, name: 'Ocean Vibes' }, { id: 22, name: 'Seeing Red' }];
    const palettes = [
      { id: 6, name: 'Pal1', projects_id: 20, color1: '#baddc3', color2: '#ff6767', color3: '#ffe596', color4: '#b3e6ff', color5: '#632ad7' },
      { id: 7, name: 'Pal2', projects_id: 20, color1: '#ab7146', color2: '#b3e6ff', color3: '#41406d', color4: '#b3e66tr', color5: '#41406d' },
      { id: 8, name: 'Pal3', projects_id: 22, color1: '#993f42', color2: '#d8ac97', color3: '#4e6cc4', color4: '#b4r35', color5: '#41406d' }
    ]
    wrapper = shallow(<PaletteForm
      projects={projects}
      color1={'#baddc3'}
      color2={'#d8ac97'}
      color3={'#993f42'}
      color4={'#b4r35'}
      color5={'#ff6767'}
      key='palette-form'
    />)
  });

  it('should match the snapshot with all data passed in', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('should update paletteName in state when handleChange is called', () => {
    const mockEvent = {
      preventDefault: jest.fn(),
      target: {
        name: 'paletteName',
        value: 'Fish'
      }
    }

    wrapper.instance().handleChange(mockEvent)

    expect(wrapper.state('paletteName')).toEqual('Fish')
  })

  it('should update projectName in state when handleClick is called', () => {
    const mockEvent = {
      preventDefault: jest.fn(),
      target: {
        name: 'projectName',
        value: 'Sea',
        innerText: 'Sea'
      }
    }

    wrapper.instance().getProjectNames()
    wrapper.instance().handleClick(mockEvent)

    expect(wrapper.state('projectName')).toEqual('Sea')
  })

  it('should reset paletteName, projectName, and error in state when resetAllInputs is called', () => {
    const defaultState = {
      paletteName: 'Fish',
      projectName: 'Sea',
      error: 'Maybe an error here',
      savingPalette: false,
      buttonText: 'Select Project'
    }

    const expectedState = {
      paletteName: '',
      projectName: '',
      error: '',
      savingPalette: false,
      buttonText: 'Select Project'
    }

    wrapper.instance().setState(defaultState)
    wrapper.instance().resetAllInputs()

    expect(wrapper.state()).toEqual(expectedState)
  })

  it('should call submitForm when the save palette button is clicked', () => {
    const mockEvent = { preventDefault: jest.fn() };
    wrapper.instance().submitForm = jest.fn();
    const defaultState = {
      paletteName: '',
      projectName: '',
      error: '',
      savingPalette: true
    }

    wrapper.instance().setState(defaultState)
    wrapper.find('.save-palette').simulate('click', mockEvent);

    expect(wrapper.instance().submitForm).toHaveBeenCalled();
  })
});
