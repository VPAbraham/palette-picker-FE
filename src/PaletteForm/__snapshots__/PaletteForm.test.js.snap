// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProjectForm should match the snapshot with all data passed in 1`] = `
<form
  className="palette-form"
>
  <div
    className="menu-items"
  >
    <img
      alt="white plus symbol"
      className="modalPlus"
      onClick={[Function]}
      src="plus_white.svg"
    />
    <h1>
      SAVE PALETTE
    </h1>
  </div>
</form>
`;
